Sources/SwiftDoubleRatchet/Num.swift

```diff
@@ -25,17 +25,13 @@ extension UInt64 {
         return Data(ret)
     }
     
-    init(array: [UInt8]) throws {
-        if array.count != 8 {
+    init<C: Collection>(collection: C) throws where C.Element == UInt8 {
+        guard collection.count == 8 else {
             throw UInt64DecodeError.invalidLength
         }
-        var value: UInt64 = 0
-        for index in 0...7 {
-            let element = array[index]
-            let offset = 8 * UInt64(index)
-            let mask: UInt64 = UInt64(element) << offset
-            value = value | mask
-        }
-        self = .init(littleEndian: value)
```

Comment:
It's a bit nicer to make this initialiser generic, because then you don't have to cast the data slices to `Array` every time.